extensions/positron-r/src/session.ts

```diff
@@ -737,6 +737,21 @@ export class RSession implements positron.LanguageRuntimeSession, vscode.Disposa
 		}
 	}
 
+	public async activateLsp() {
+		// Start LSP for the foreground session only if its been previously stopped
+		if (this._lsp.state === LspState.stopped) {
+			const port = await this.adapterApi!.findAvailablePort([], 25);
+			this._lsp.activate(port, this.context);
+		}
+	}
+
```

Comment:
Also, as @lionel- mentioned earlier right now `startLsp()` is called from within `this._queue.add()` to help with request ordering. I imagine this probably needs to be in the queue too? But I'm not entirely sure.